pkg/scheduler/frameworkext/services/services.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"net/http"
+	"sort"
+	"sync"
+	"sync/atomic"
+
+	"github.com/gin-gonic/gin"
+	"k8s.io/kubernetes/pkg/scheduler"
+
+	"github.com/koordinator-sh/koordinator/pkg/descheduler/framework"
+)
+
+const (
+	servicesBaseRelativePath       = "/apis/v1/"
+	pluginServicesBaseRelativePath = servicesBaseRelativePath + "plugins"
+)
+
+var once sync.Once
+var engine atomic.Value
+
+type mux interface {
+	Handle(string, http.Handler)
+	HandlePrefix(string, http.Handler)
+}
+
+func InstallAPIHandler(mux mux, e *Engine, sched *scheduler.Scheduler, isLeader func() bool) {
```

Comment:
can you provide a user manual to show how to really use in real envirement, such curl ip:port/method? how to get port for example? thx~